Intersect.Client/Entities/Entity.cs

```diff
@@ -1878,6 +1878,64 @@ protected virtual bool TryGetAnimationTexture(string textureName, SpriteAnimatio
             return texture != default;
         }
 
+        /// <summary>
+        /// <para>Returns the direction to a player's selected target.
+        /// Thanks to Daywalkr (Middle Ages: Online) for this !</para>
+        /// </summary>
+        /// <param name="en">entity's target</param>
+        /// <returns>direction to player's selected target</returns>
+        protected byte DirectionToTarget(Entity en)
+        {
+            byte newDir = Dir;
+
+            if (en == null)
+            {
+                return newDir;
+            }
+
+            int originY = Y;
+            int originX = X;
+            int targetY = en.Y;
+            int targetX = en.X;
+
+            if (en.MapInstance.Id != MapInstance.Id)
+            {
+                if (en.MapInstance.GridY < MapInstance.GridY)
+                {
+                    originY += Options.MapHeight - 1;
+                }
+                else if (en.MapInstance.GridY > MapInstance.GridY)
+                {
+                    targetY += Options.MapHeight - 1;
+                }
+
+                if (en.MapInstance.GridX < MapInstance.GridX)
+                {
+                    originX += Options.MapWidth - 1;
+                }
+                else if (en.MapInstance.GridX > MapInstance.GridX)
+                {
+                    targetX += (Options.MapWidth - 1);
+                }
+            }
+
+            var yOffset = originY - targetY;
+            var xOffset = originX - targetX;
+
+            if ((Math.Abs(yOffset) - Math.Abs(xOffset)) > 0)
+            {
+                newDir = (Math.Abs(yOffset) != 0) ? (yOffset > 0) ? (byte)Directions.Up : (byte)Directions.Down :
+                    (xOffset > 0) ? (byte)Directions.Left : (byte)Directions.Right;
+            }
+            else
+            {
+                newDir = (Math.Abs(xOffset) != 0) ? (xOffset > 0) ? (byte)Directions.Left : (byte)Directions.Right :
+                    (yOffset > 0) ? (byte)Directions.Up : (byte)Directions.Down;
+            }
```

Comment:
So on further thought I think there is value for switches here but not quite the way you had done before.

I would instead replace the entire if statement with

```csharp
var <x/y>Dir = <x/y>Offset == 0 ? newDir : (<x/y>Offset < 0 ? (byte)Direction.<direction> : (byte)Direction.<direction>);
var <x/y>Dir = <x/y>Offset == 0 ? newDir : (<x/y>Offset < 0 ? (byte)Direction.<direction> : (byte)Direction.<direction>);
newDir = Math.abs(yOffset) > Math.abs(xOffset) ? yDir : xDir;
```

Reply:
This isn't really much more readable, but it is slightly more readable than what you would have to change the code to in order to fix the case where yOffset isn't 0 and xOffset is (currently Right instead of current newDir), and where xOffset isn't 0 but yOffset is (currently it's Down instead of the current newDir) -- which this change does fix.